### DISCUSSION TITLE
How to use the intersect query ?

### DISCUSSION THREAD

**Discussion by usha-chawla-infosys** [ujjwalguptaofficial/JsStore#315] [Q&A] [answered]
Hi All,

I am trying to use intersect in the typescript code but I am getting an error.
To use intersect, I created 2 queries and added them as parameters to the intersect query. But it does not work for me. Kindly suggest what needs to be done here. PFB the code:

Error: Argument of type '{ queries: Promise<{}[]>[]; }' is not assignable to parameter of type 'SelectQuery[]'.
  Object literal may only specify known properties, and 'queries' does not exist in type 'SelectQuery[]'.

 const queryProd1 =  connection.select({
      from: "Contents",
      where: {
        productName: "Bread"
      }
    });

    const queryProd2 = connection.select({
      from: "Contents",
      where: {
        productType: "Bakes"
      }
    });

    var results = connection.intersect({
      queries:[queryProd1,queryProd2]    
  })

**Comment by ujjwalguptaofficial** [comment, marked as answer]
You just need to specify the query not results like this - 

```
var query1=  {
    from: 'Products',
    where: {
        price: {
            '>': 10
        }
    }
};
var query2 = {
    from: 'Products',
    where: {
        price: {
            '>': 50
        }
    }
};
var results = await connection.intersect({
    queries:[query1,query2]    
})
```

**Reply by ujjwalguptaofficial** [reply]
In your code - you are using `connection.select` but need to specify the query only. JsStore takes the query - execute them and then merge them.

**Reply by usha-chawla-infosys** [reply]
Thanks, Ujjwal for replying. Can I please know if I should use the connection with web worker or without web worker? Currently, I am using a connection with web worker.

Because I tried your code and I am still getting this error :

Argument of type '{ queries: SelectQuery[]; }' is not assignable to parameter of type 'SelectQuery[]'.
  Object literal may only specify known properties, and 'queries' does not exist in type 'SelectQuery[]'.

**Reply by ujjwalguptaofficial** [reply]
can you share screenshot of your code ?

**Reply by ujjwalguptaofficial** [reply]
worker or without worker both should work - the problem is with the query not connection

**Reply by usha-chawla-infosys** [reply]
PFA the screenshots.

<img width="430" alt="Code " src="https://user-images.githubusercontent.com/59990263/216601781-8bedbb4a-1d47-428c-8862-b848893806a8.png">
<img width="731" alt="Base service class" src="https://user-images.githubusercontent.com/59990263/216601801-81749cfe-d57e-4e6b-b04c-dc2900288096.png">
<img width="798" alt="Indexed DB service class" src="https://user-images.githubusercontent.com/59990263/216601819-3a3d92e8-b482-4591-9478-3339444ba64b.png">

**Reply by ujjwalguptaofficial** [reply]
convert array into any and issue should be gone - 

```
{
  queries: [qry1,qry2] as any
}
```
Let me know if it works

**Reply by usha-chawla-infosys** [reply]
Sorry, this did not work. I am still getting this error : 
Argument of type '{ queries: any; }' is not assignable to parameter of type 'SelectQuery[]'.
  Object literal may only specify known properties, and 'queries' does not exist in type 'SelectQuery[]'.

**Reply by ujjwalguptaofficial** [reply]
Hey Usha - drop me a message at my linked in : https://www.linkedin.com/in/ujjwalkrgupta/ - We can get on a call and then fix this.
